Sources/Core/AWSClientRuntime/AWSClientConfiguration.swift

```diff
@@ -240,9 +269,20 @@ extension AWSClientConfiguration {
             profileName: nil,
             fileBasedConfig: fileBasedConfig
         )
+        var resolvedAuthSchemes: [ClientRuntime.AuthScheme] = []
+        if let authSchemes = authSchemes {
+            resolvedAuthSchemes = authSchemes
+        } else {
+            resolvedAuthSchemes.append(SigV4AuthScheme())
+            // These AWS services use SigV4A auth scheme.
+            if ["S3", "EventBridge", "CloudFront KeyValueStore"].contains(serviceSpecific?.serviceName) {
```

Comment:
same Q as above + 
nit: is there a way to reduce code duplication? If this is more readable I am good with it though

Reply:
With the new config / plugin structure, the auth schemes defaults provider that returns auth schemes (supported by the SDK) handles this resolve logic  now. Thanks to that, this duplication is gone now (will be gone-gone when this file is deleted down the line).